docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+
+* **Message Queue**
+
+  RabbitMQ instance for storage of requests received by the RC
+  allowing them to be polled by a consumer.
+  Requests for which acknowledgement is received upon polling
+  will be deleted from the queue,
+  while the ones without acknowledgement will be requeued.
+
+* **Database (optional, to be implemented at stage 3)**
+
+  Relational DB storing the state of the DP.
+
+  DB allows to keep track of grants and CBSDs configuration limiting redundant CBSD - SAS connections.
+
+## Component Interactions
+
+* **Protocol Controller**
+
+  PC will pass the requests it receives from CBSDs to the RC over HTTPS.
+  
+  PC will pass the responses that are coming from RC back to the respective CBSDs
+  over the protocol that they are able to understand and the PC is implementing.
+
+* **Radio Controller**
+
+  RC receives requests from the PC over HTTPs and puts them in
+  the queue over AMQP.
+
+  When the response for the request comes back from CC,
+  RC will forward it over HTTPS to the appropriate PC.
+
+* **Configuration Controller**
+
+  CC communicates with the queue over AMQP for request polling.
+  
+  CC communicates with SAS (external) over HTTPS for posting bulked requests and with the RC over HTTPS for forwarding the SAS response.
+
+* **Message Queue**
+
+  Treating RC as the publisher and CC as the consumer.
+
+* **Database**
+
+  Exchanging information about CBSDs and grants with the CC.
+
+## Solution scaling
+
+The Domain Proxy will be split into separate services in separate containers.
+
+Every service will work in HA mode to provide service even in failover scenarios.
+
+Components such as database and queue will use cluster solutions
+delivered by database and queue providers.
+
+All other components will be stateless Python based services
+which do not need synchronization between them
+so it is not vital to create sync mechanisms.
+Python scripts will take over processing of data
+when one of the containers fail.
+When a container that processes requests received from RabbitMQ fails
+then RabbitMQ will not receive confirmation of received requests from the queue
+and will queue requests again.
+The only state will be kept in the database.
+
+## Batching of messages
+
+Batching of messages allows to limit the number of connections to the SAS servers
+which is useful in the case when a number of CBSDs send requests at once
+(e.g. heartbeat requests are sent continuously).
+Batching it limits a chance of failure on sending requests to the SAS,
+because SAS has to process only a minimum number of requests
+instead of numerous requests sent one by one.
+In large infrastructures it may reduce the cost of SAS usage
+and limit the chance of running out of requests per given time
+(for Google SAS server it is 50,000 requests per day and 10 queries per second).
+
+Keeping the state is not necessary on the early stage of the project
+when Domain Proxy supports only Passive Mode
```

Comment:
I'm unable to understand the usecase for passive mode. Why would we ever need passive mode ?

Reply:
The main use case is the first bullet in Rationale portion of the doc. Batching and limiting unique connection endpoints outbound from a deployment network.

Reply:
Passive mode is let's say the first milestone - as a result of which we can set up end-to-end CBSD to SAS connection through the DP. Without advanced use cases, which are :
 - Configure DP to request additional Grants for specific CBSDs, to get back up grant which can be used in case of any issues with the main grant.
 - some other which @jpad-freedomfi  can outline